File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegate.java
package me.ykrank.s1next.view.internal;

import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import me.ykrank.s1next.viewmodel.LoadingViewModel;

/**
 * This class represents a delegate which you can bind
 * {@link LoadingViewModel} to different {@link android.databinding.ViewDataBinding}s
 * in implementation.
 */
public interface LoadingViewModelBindingDelegate {

    View getRootView();

    SwipeRefreshLayout getSwipeRefreshLayout();

    /**
     * This {@link RecyclerView} should always set a
     * {@link me.ykrank.s1next.view.adapter.BaseRecyclerViewAdapter}
     * implementation, otherwise we can not use
     * {@link me.ykrank.s1next.view.adapter.BaseRecyclerViewAdapter#setHasProgress(boolean)}.
     */
    RecyclerView getRecyclerView();

    void setLoadingViewModel(LoadingViewModel loadingViewModel);
}

File: library/src/main/java/com/github/ykrank/androidtools/ui/internal/LoadingViewModelBindingDelegate.kt
package com.github.ykrank.androidtools.ui.internal

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.github.ykrank.androidtools.ui.vm.LoadingViewModel

/**
 * This class represents a delegate which you can bind
 * [LoadingViewModel] to different [androidx.databinding.ViewDataBinding]s
 * in implementation.
 */
interface LoadingViewModelBindingDelegate {
    val rootView: View
    val swipeRefreshLayout: SwipeRefreshLayout

    /**
     * This [RecyclerView] should always set a
     * [LibBaseRecyclerViewAdapter]
     * implementation, otherwise we can not use
     * [LibBaseRecyclerViewAdapter.setHasProgress].
     */
    val recyclerView: RecyclerView

    val hintView: TextView
    fun setLoadingViewModel(loadingViewModel: LoadingViewModel)
}
